float64[7] Torque
float64[16] EndPos